packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -1,5 +1,84 @@
-function reducer(state = {}): typeof state {
-  return state;
+import { AnyAction, Dispatch } from 'redux';
+import { getAggregations } from './../utlis/aggregations';
+import { getQueries } from './../utlis/queries';
+
+enum actions {
+  ITEMS_FETCHED = 'itemsFetched',
+}
+
+type StateActions = {
+  type: actions.ITEMS_FETCHED;
+  payload: Item[];
+};
+
+export type Item = {
+  id: string;
+  lastModified: number;
+  name: string;
+  namespace: string;
+  type: 'query' | 'aggregation';
+};
+
+export type State = {
+  loading: boolean;
+  items: Item[];
+};
+
+const INITIAL_STATE: State = {
+  loading: true,
+  items: [],
+};
+
+function reducer(
+  state = INITIAL_STATE,
+  action: StateActions | AnyAction
+): State {
+  const newState = { ...state };
+  if (action.type === actions.ITEMS_FETCHED) {
+    newState.items = action.payload;
+    newState.loading = false;
+  }
+  return newState;
 }
 
+export const fetchItems = () => {
+  return async (dispatch: Dispatch<StateActions>): Promise<void> => {
+    const payload = await Promise.allSettled([
+      getAggregationItems(),
+      getQueryItems(),
+    ]);
+    dispatch({
+      type: actions.ITEMS_FETCHED,
+      payload: payload
+        .map((result: PromiseSettledResult<Item[]>) =>
+          result.status === 'fulfilled' ? result.value : []
+        )
+        .filter(Boolean)
```

Comment:
Wouldn't the value on fulfilled always be present and be truthy? I think you can remove this filter if that's the case (or return something that is `falsy` instead of an array of you want to filter it), otherwise it's a bit redundant

```suggestion
        .map((result: PromiseSettledResult<Item[]>) =>
          result.status === 'fulfilled' ? result.value : []
        )
```